blog/content/introtogonum.md

```diff
@@ -0,0 +1,145 @@
+# The Gonum Numeric Computing Packages
+
+## Gonum -- The Basics
+
+Gonum is a set of packages designed to make writing numeric and scientific
+algorithms productive, performant, and scalable.
+
+Gonum is a set of packages written in the 
+[Go programming language](https://www.golang.org) (more on Go later), and is
+not a language on its own.
+This makes Gonum more similar to [numpy](https://www.numpy.org) and
+[scipy](https://www.scipy.org), libraries built on top of
+[python](https://www.python.org), and different from
+[Julia](https://julialang.org/) and [Matlab](https://www.mathworks.com/)
+which are full programming languages.
+
+Gonum contains libraries for [matrices and linear algebra](http://godoc.org/gonum.org/v1/gonum/mat);
+[statistics](http://godoc.org/gonum.org/v1/gonum/stat), 
+[probability](http://godoc.org/gonum.org/v1/gonum/stat/distuv) 
+[distributions]((http://godoc.org/gonum.org/v1/gonum/stat/distmv), 
+and [sampling](http://godoc.org/gonum.org/v1/gonum/stat/sampleuv); tools for
+[function differentiation](http://godoc.org/gonum.org/v1/gonum/diff/fd), 
+[integration](http://godoc.org/gonum.org/v1/gonum/integrate/quad),
+and [optimization](http://godoc.org/gonum.org/v1/gonum/optimize);
+[network](http://godoc.org/gonum.org/v1/gonum/graph) creation and analysis; and more.
+
+## Gonum Philosophy
+The Gonum packages are designed to be simple, efficient and composable.
+There is a tension, especially in scientific computing, between ease and simplicity.
+Gonum, [like the Go language itself](https://commandcenter.blogspot.com/2012/06/less-is-exponentially-more.html),
+aims first to provide reliable and predictable code, while trying to maintain
+ease-of-use.
+Gonum is different from other popular scientific libraries, in that the Go language
+does not allow operator or method overloading.
+On the surface, this often means that the same algorithm requires a few more
+lines of code in Go.
+We fully recognize that sexy one-liners are rare with Gonum.
+
+We accept this cost because we believe it comes with an enormous benefit.
+It is easy to identify and explore the exact code segments being executed, and
+the Gonum function and method signatures make it clear what is happening.
+Science is built on understanding how an experiment was performed and understanding
+what the results are.
+We believe scientific code should be no different.
+In fact, Gonum is partly distinguished from other libraries in that the entire
+codebase can be run with pure Go -- you don't need to know another programming
+language to read Gonum code (that said, it is also easy to call into existing C libraries).
+
+This fundamental philosophy has many follow-on benefits.
+We find that one-liners often obscure the underlying operations, leading to
```

Comment:
I tried a rephrase, but it's probably not enough.

My point is that in Matlab, you can implement Mahalanobis distance as `(mu-x)'*Sigma^-1*(mu-x)`. The Gonum code is much more elaborate. However, the matlab code disguises that the transpose is an allocation and a copy, it computes the explicit inverse, and disguises the temporary matrices that get created to perform each operation. The idiomatic `Sigma\(mu-x)` is better, but still ignores that Sigma should be a SPD matrix, and so one should use the Cholesky decomposition and related Solve routine. Point being, there are a lot of hidden costs to the simple-looking code.